RecoTauTag/TauTagTools/src/TauTagTools.cc

```diff
@@ -101,64 +102,72 @@ void replaceSubStr(string& s,const string& oldSubStr,const string& newSubStr){
     return filteredTracks;
   }
 
-  std::vector<reco::PFCandidatePtr> filteredPFChargedHadrCandsByNumTrkHits(const std::vector<reco::PFCandidatePtr>& theInitialPFCands, int ChargedHadrCand_tkminTrackerHitsn){
-    std::vector<reco::PFCandidatePtr> filteredPFChargedHadrCands;
-    for(std::vector<reco::PFCandidatePtr>::const_iterator iPFCand=theInitialPFCands.begin();iPFCand!=theInitialPFCands.end();iPFCand++){
-      if (PFCandidate::ParticleType((**iPFCand).particleId())==PFCandidate::h  || PFCandidate::ParticleType((**iPFCand).particleId())==PFCandidate::mu || PFCandidate::ParticleType((**iPFCand).particleId())==PFCandidate::e){
+  std::vector<reco::CandidatePtr> filteredPFChargedHadrCandsByNumTrkHits(const std::vector<reco::CandidatePtr>& theInitialPFCands, int ChargedHadrCand_tkminTrackerHitsn){
+    std::vector<reco::CandidatePtr> filteredPFChargedHadrCands;
+    for(std::vector<reco::CandidatePtr>::const_iterator iPFCand=theInitialPFCands.begin();iPFCand!=theInitialPFCands.end();iPFCand++){
+      if (std::abs((*iPFCand)->pdgId()) == 211 || std::abs((*iPFCand)->pdgId()) == 13 || std::abs((*iPFCand)->pdgId()) == 11){
```

Comment:
this code apparently doesn't work with non-PF and it is also not a part of some generic code where Cand can work.
Please revert to using PF

Reply:
With this being tools used only by outdated algorithms, I addressed the comment by using the bestTrack() methods, which we should most likely also use in general.